graylog2-web-interface/src/stores/metrics/MetricsStore.js

```diff
@@ -43,46 +42,41 @@ const MetricsStore = Reflux.createStore({
       return;
     }
     const metricsToFetch = {};
+
+    // First collect all node metric registrations
     Object.keys(this.registrations)
-      .filter((nodeId) => Object.keys(this.registrations[nodeId]).length > 0)
-      .forEach(nodeId => {
-        metricsToFetch[nodeId] = Object.keys(this.registrations[nodeId]).filter(metricName => this.registrations[nodeId][metricName] > 0);
-      });
-    const globalMetrics = Object.keys(this.globalRegistrations).filter(metricName => this.globalRegistrations[metricName] > 0);
-
-    if (this.nodes) {
-      Object.keys(this.nodes).forEach(nodeId => {
-        globalMetrics.forEach(metricName => {
-          if (!metricsToFetch[nodeId]) {
-            metricsToFetch[nodeId] = [];
-          }
-          if (metricsToFetch[nodeId].indexOf(metricName) == -1) {
-            metricsToFetch[nodeId].push(metricName);
-          }
-        });
+      .filter((nodeId) => Object.keys(this.registrations[nodeId].length > 0))
+      .forEach((nodeId) => {
+        Object.keys(this.registrations[nodeId])
+          .filter((metricName) => this.registrations[nodeId][metricName] > 0)
+          .forEach((metricName) => {
+            metricsToFetch[metricName] = 1;
+          });
       });
-    }
 
-    const promises = Object.keys(metricsToFetch)
-      .map((nodeId) => {
-        const url = URLUtils.qualifyUrl(ApiRoutes.ClusterMetricsApiController.multiple(nodeId).url);
-        const body = {metrics: metricsToFetch[nodeId]};
-        return fetch('POST', url, body).then((response) => {
-          const metrics = {};
-          response.metrics.forEach((metric) => metrics[metric.full_name] = metric);
-          return {nodeId: nodeId, metrics: metrics};
-        });
+    // Then collect all global metric registrations
+    Object.keys(this.globalRegistrations)
+      .filter((metricName) => this.globalRegistrations[metricName] > 0)
+      .forEach((metricName) => {
+        metricsToFetch[metricName] = 1;
       });
 
-    const promise = this._allResults(promises).then((responses) => {
+    const url = URLUtils.qualifyUrl(ApiRoutes.ClusterMetricsApiController.multipleAllNodes().url);
```

Comment:
Do we really want to get all metrics for all nodes, even when we only need metrics from one of them? Going to System -> Nodes -> Node fetches metrics about that node, and all other nodes in the cluster, even though we are not using them.